src/components/UTIL_FormField.component

```diff
@@ -56,15 +56,19 @@
         rendered="{!($ObjectType[sObjType].fields[field].updateable
                     || $ObjectType[sObjType].fields[field].createable && sObj['Id']=='') && $ObjectType[sObjType].fields[field].type != 'boolean'}">        
         <apex:outputLabel for="inputX" styleClass="slds-form-element__label" rendered="{!showLabel == true && ftype!='boolean'}" id="lblInputField">
-            <apex:outputText rendered="{!required}" id="lblInputFieldRequired">
+            <apex:outputText rendered="{!required || NOT($ObjectType[sObjType].fields[field].nillable)}" id="lblInputFieldRequired">
```

Comment:
@jjbennett, the `required` attribute cannot be set to be empty because even if it is empty, null, 0, ... it is considered as `false`.  So, in order to check if the optional `required` attribute needs to be set, we need to check if the field is not `nillable`. 
If this approach does not work, the other (only) option is to change all pages using the `UTIL_FormField` to pass in required value.

Reply:
@RadSFDeveloper Instead of setting a default on line 6, could we create an apex:variable and check if the attribute has a value and if it does use it, if it does not then use the value from nillable then use the new var instead of the required attribute?

Reply:
@jjbennett ,  I have removed the `default` and created a new variable to check if the `required` is not set. However, since `required` is a `Boolean`, then any passed value that is not `true` is `falsey`, ie the returned value is `false`. Do you see anything wrong with the following code that does not recognize the `required` attribute is empty?

`
    <apex:variable var="isRequired"
        value="{!IF(required == null, NOT($ObjectType[sObjType].fields[field].nillable), required)}"
    />`